### ISSUE TITLE
different behaviour of `unique()` in 1.9.8

### ISSUE THREAD

**Issue by manuelreif** [Rdatatable/data.table#1966]
Hi!

I am using `data.table` 1.9.8 and I noticed a different behaviour when I use `unique()` compared to the 1.9.6 version.

```r
# version 1.9.8
library(data.table)
set.seed(123)
DT <- data.table(x=c(1,1,1,2,2,2,3,3,3),y=rnorm(9))
setkey(DT,x)
unique(DT)

   x           y
1: 1 -0.56047565
2: 1 -0.23017749
3: 1  1.55870831
4: 2  0.07050839
5: 2  0.12928774
6: 2  1.71506499
7: 3  0.46091621
8: 3 -1.26506123
9: 3 -0.68685285

# version 1.9.6
# result:

   x           y
1: 1 -0.56047565
2: 2  0.07050839
3: 3  0.46091621
```

The second ('old') result is more what I expected. To get the same result as 1.9.6, I used the `by` argument, which should not be necessary following the instructions in the `unique.data.table` help file.

```r
unique(DT,by="x")
   x           y
1: 1 -0.56047565
2: 2  0.07050839
3: 3  0.46091621
```

Thank you!
Manuel

**Comment by DavidArenburg** [comment]
Please read `?unique`

**Comment by franknarf1** [comment]
The NEWS file covers the change in its first item: https://github.com/Rdatatable/data.table/blob/master/NEWS.md#changes-in-v198--on-cran-25-nov-2016 

Fyi, support instructions are here: https://github.com/Rdatatable/data.table/wiki/Support

@DavidArenburg I think the OP is right about the docs. Looking at my 1.9.8 docs (and not seeing anything in the NEWS about further changes), I see:

> unique returns a data.table with duplicated rows (by key) removed, or (when no key) duplicated rows by all columns removed. 
> [...]
> by	
> character or integer vector indicating which combinations of columns form x to use for uniqueness checks. Defaults to key(x)) which, by default, only uses the keyed columns. by=NULL uses all columns and acts like the analogous data.frame methods.
> [...]
> Any combination of columns can be used to test for uniqueness (not just the key columns) and are specified via the by parameter. To get the analagous data.frame functionality, set by to NULL.

**Comment by DavidArenburg** [comment]
@franknarf1 I don't have 1.9.8 in-front of me and not sure why would anyone use a version that was released for barely a week. The 1.10.1 docs explicitly say

> character or integer vector indicating which combinations of columns from x to use for uniqueness checks. By default all columns are are being used. The was changed recently for consistency to data.frame methods. In version < 1.9.8 default was key(x).

**Comment by franknarf1** [comment]
@DavidArenburg Okay, just upgraded to 1.10.0 and I see it's been fixed. Kind of surprised it's not a NEWS item, but not a big deal.

@manuelreif For what it's worth, I guess this is why the support instructions ask that you update to the latest version before reporting.

**Comment by jangorecki** [comment]
Yes, we missed update of those docs for 1.9.8. It was fixed in 1.9.9 and is now released in CRAN in 1.10.0. Still the NEWS.md was quite clear on that, so be sure to read breaking changes before upgrading.

**Comment by nachti** [comment]
NOTE: Sometimes your system is maintained by somebody else (e.g. IT Department) and you cannot choose which version you use so you have to trust in the help files and it's maybe not possible to upgrade packages yourself - even if it's inconvenient. Therefore thanks @manuelreif for reporting and to the developers for fixing the help file.
